src/Infrastructure/Captcha/GoogleRecaptchaV3Service.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.Extensions.Configuration;
+using System.Runtime.Serialization.Json;
+using System.Web;
+
+namespace FSH.WebApi.Infrastructure.Captcha;
+
+public class GoogleRecaptchaV3Service : ICaptchaService
+{
+    public HttpClient _httpClient { get; set; }
+
+    public string RequestPath { get; set; }
+    public string RequestSecret { get; set; }
+    public string RequestResponse { get; set; }
+    public string RequestRemoteip { get; set; }
+
+    public GResponseModel Response { get; set; }
+
+    public HttpRequestException HttpReqException { get; set; }
+
+    public Exception GeneralException { get; set; }
+
+
+    public GoogleRecaptchaV3Service(HttpClient httpClient, IConfiguration config)
+    {
+        _httpClient = httpClient;
+        var settings = config.GetSection(nameof(CaptchaSettings)).Get<CaptchaSettings>();
+
+        RequestSecret = settings.GoogleRecaptchaV3.Secret;
+        RequestPath = settings.GoogleRecaptchaV3.ApiUrl;
+    }
+
+    public void InitializeRequest(string RecaptchaToken, string RemoteIpAddress)
+    {
+        RequestRemoteip = RemoteIpAddress;
+        RequestResponse = RecaptchaToken;
+    }
+
+    public async Task<bool> Execute()
+    {
+        // Notes on error handling:
+        // Google will pass back a 200 Status Ok response if no network or server errors occur.
+        // If there are errors in on the "business" level, they will be coded in an array;
+        // CaptchaRequestException is for these types of errors.
+
+        // CaptchaRequestException and multiple catches are used to help seperate the concerns of 
+        //  a) an HttpRequest 400+ status code
+        //  b) an error at the "business" level 
+        //  c) an unpredicted error that can only be handled generically.
+
+        // It might be worthwhile to implement a "user error message" property in this class so the
+        // calling procedure can decide what, if anything besides a server error, to return to the 
+        // client and any client handling from there on.
+        try
+        {
+
+            // Don't to forget to invoke any loggers in the logic below.
+
+            // formulate request
+            string builtURL = $"{RequestPath}?" +
+                HttpUtility.UrlPathEncode($"secret={RequestSecret}&response={RequestResponse}&remoteip={RequestRemoteip}");
```

Comment:
There is a UriBuilder class for stuff like this.

Also: the base url can be specified when you register the httpclient during startup. Then you only have to add the parameters here.